rasa/importers/importer.py

```diff
@@ -0,0 +1,247 @@
+
+        raise NotImplementedError()
+
+    async def get_config(self) -> Dict:
+        """Retrieves the configuration which should be used for the training."""
+        raise NotImplementedError()
+
+    async def get_nlu_data(self, language: Optional[Text] = "en") -> TrainingData:
+        """Retrieves the nlu training data which should be used for the training."""
+        raise NotImplementedError()
+
+    @staticmethod
+    def load_from_config(
+        config_path: Text,
+        domain_path: Optional[Text] = None,
+        training_data_paths: Optional[List[Text]] = None,
+        load_only_nlu_data: bool = False,
+    ) -> "TrainingFileImporter":
+        """Loads a `TrainingFileImporter` instance from a configuration file."""
+
+        config = io_utils.read_config_file(config_path)
+        return TrainingFileImporter.load_from_dict(
+            config, config_path, domain_path, training_data_paths, load_only_nlu_data
+        )
+
+    @staticmethod
+    def load_from_dict(
+        config: Optional[Dict],
+        config_path: Text,
+        domain_path: Optional[Text] = None,
+        training_data_paths: Optional[List[Text]] = None,
+        load_only_nlu_data: bool = False,
+    ) -> "TrainingFileImporter":
+        """Loads a `TrainingFileImporter` instance from a dictionary."""
+        config = config or {}
+        importers = config.get("importers", [])
+        importers = [
+            TrainingFileImporter._importer_from_dict(
+                importer, config_path, domain_path, training_data_paths
+            )
+            for importer in importers
+        ]
+        importers = [importer for importer in importers if importer]
+
+        if not importers:
+            importers = [
+                SimpleFileImporter(config_path, domain_path, training_data_paths)
+            ]
+
+        importer = CombinedFileImporter(importers)
+        if load_only_nlu_data:
+            importer = NluFileImporter(importer)
+
+        return importer
+
+    @staticmethod
+    def _importer_from_dict(
+        importer_config: Dict,
+        config_path: Text,
+        domain_path: Optional[Text] = None,
+        training_data_paths: Optional[List[Text]] = None,
+    ) -> Optional["TrainingFileImporter"]:
+        from rasa.importers.skill import SkillSelector
+
+        module_path = importer_config.pop("name", None)
+        if module_path == SimpleFileImporter.__name__:
+            importer_class = SimpleFileImporter
+        elif module_path == SkillSelector.__name__:
+            importer_class = SkillSelector
+        else:
+            try:
+                importer_class = common_utils.class_from_module_path(module_path)
+            except (AttributeError, ImportError):
+                logging.warning("Importer '{}' not found.".format(module_path))
+                return None
+
+        import rasa.cli.utils as cli_utils
+
+        constructor_arguments = cli_utils.minimal_kwargs(
```

Comment:
I guess, we should move this method (`minimal_kwargs`) to `utils.common` as it is not only used in the cli package. But should maybe happen in a different PR.